backend/getDisplayName.js

```diff
@@ -11,12 +11,12 @@
 'use strict';
 
 const FB_MODULE_RE = /^(.*) \[from (.*)\]$/;
-const cachedDisplayNames = new WeakMap();
+const cachedDisplayNames: WeakMap<Function, string> = new WeakMap();
 
 function getDisplayName(type: Function): string {
-  if (cachedDisplayNames.has(type)) {
-    // $FlowFixMe From the upgrade to Flow 64
-    return cachedDisplayNames.get(type);
+  const nameFromCache = cachedDisplayNames.get(type);
+  if (nameFromCache != null) {
```

Comment:
`typeof !== 'string'` check might be faster

Reply:
hmm, looks like it might be a wash ¯\_(ツ)_/¯

```js
const Benchmark = require('benchmark');

function getCache() {
  const cachedDisplayNames = new WeakMap();
  cachedDisplayNames.set(function exists() {}, 'hello');
  return cachedDisplayNames;
}

function getSuite() {
  return new Benchmark.Suite()
    .on('cycle', function(event) {
      console.log(String(event.target));
    })
    .on('complete', function() {
      console.log('Fastest is ' + this.filter('fastest').map('name'));
    });
}


// add tests
getSuite()
  .add('typeof with cached name', function() {
    typeof getCache().get('exists') === 'string';
  })
  .add('!= null with cached name', function() {
    getCache().get('exists') != null;
  })
  .run();

console.log('------------')

getSuite()
  .add('typeof with cache miss', function() {
    getCache().get('dne') === 'string';
  })
  .add('!= null with cache miss', function() {
    getCache().get('dne') != null;
  })
  .run();
```

results in

```
typeof with cached name x 4,954,129 ops/sec ±1.78% (85 runs sampled)
!= null with cached name x 5,012,633 ops/sec ±1.68% (90 runs sampled)
Fastest is != null with cached name,typeof with cached name
------------
typeof with cache miss x 5,012,751 ops/sec ±1.60% (89 runs sampled)
!= null with cache miss x 4,928,582 ops/sec ±2.10% (88 runs sampled)
Fastest is typeof with cache miss,!= null with cache miss
```

Reply:
Tested again without the overhead of creating the weakmap (just by comparing against a string value and undefined), and again they're within a percentage point.

I'm always happy to change this, especially if it's more in line with the rest of the code. This was habit on my part.